### PR TITLE
Update pip install to use --extra-index-url for ipex package

### PR DESCRIPTION
**Pull request by chtanch**
## Description

Update ipex-llm install from
`pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu` 
to 
`pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/`  
in readmes, code comments, and `python/llm/example/CPU/Deepspeed-AutoTP/install.sh`

Update
`pip install oneccl_bind_pt==2.1.100 -f https://developer.intel.com/ipex-whl-stable-xpu` 
to
`pip install oneccl_bind_pt==2.1.100 --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/`

### CODE REVIEW

File: docs/readthedocs/source/doc/LLM/Overview/FAQ/faq.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Please also refer to [here](https://github.com/intel-analytics/ipex-llm?tab=read
 
 ## How to Resolve Errors
 
-### Fail to install `ipex-llm` through `pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu`
+### Fail to install `ipex-llm` through `pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://developer.intel.com/ipex-whl-stable-xpu`
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/FAQ/faq.md:12]: should specify either us or cn; see https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Quickstart/install_windows_gpu.html#install-ipex-llm ot

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 2]
```diff
@@ -393,15 +393,15 @@ We recommend using [miniconda](https://docs.conda.io/en/latest/miniconda.html) t
          conda create -n llm python=3.9
          conda activate llm
 
-         pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
+         pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://developer.intel.com/ipex-whl-stable-xpu
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:396]: the link is incorrect; see https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Quickstart/install_windows_gpu.html#install-ipex-llm

**Comment by jason-dai** [comment]
> ## Description
> Update ipex-llm install from `pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu` to `pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://developer.intel.com/ipex-whl-stable-xpu` in readmes, code comments, and `python/llm/example/CPU/Deepspeed-AutoTP/install.sh`
> 
> Update `pip install oneccl_bind_pt==2.1.100 -f https://developer.intel.com/ipex-whl-stable-xpu` to `pip install oneccl_bind_pt==2.1.100 --extra-index-url https://developer.intel.com/ipex-whl-stable-xp`

The url is incorrect; see https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Quickstart/install_windows_gpu.html#install-ipex-llm

**Comment by chtanch** [comment]
Updated `https://developer.intel.com/ipex-whl-stable-xpu` to `https://pytorch-extension.intel.com/release-whl/stable/xpu/us/`

**Comment by chtanch** [comment]
Update github actions to pip install from `https://pytorch-extension.intel.com/release-whl/stable/xpu/us/` as well?

File: python/llm/README.md [hunk 3]
```diff
@@ -125,7 +125,7 @@ You may install **`ipex-llm`** on Intel GPU as follows:
 ```bash
```
Comment (jason-dai) [inline python/llm/README.md:125]: just remove this file

File: python/llm/example/CPU/Deepspeed-AutoTP/install.sh [hunk 4]
```diff
@@ -12,7 +12,7 @@ mkdir -p /opt/intel/oneccl
 mv ./_install/env /opt/intel/oneccl
 # 2. install torch and ipex
 pip install torch==2.1.0
-pip install intel_extension_for_pytorch==2.1.0 -f https://developer.intel.com/ipex-whl-stable-cpu
+pip install intel_extension_for_pytorch==2.1.0 --extra-index-url https://developer.intel.com/ipex-whl-stable-cpu
```
Comment (jason-dai) [inline python/llm/example/CPU/Deepspeed-AutoTP/install.sh:15]: this url is incorrect

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 5]
```diff
@@ -56,7 +56,7 @@ The easiest ways to install `ipex-llm` is the following commands:
 conda create -n llm python=3.9 libuv
 conda activate llm
 
-pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
+pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:59]: add CN link, either in a separate tab or in the note.

**Comment by jason-dai** [comment]
> ## Description
> Update ipex-llm install from `pip install --pre --upgrade ipex-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu` to `pip install --pre --upgrade ipex-llm[xpu] --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/` in readmes, code comments, and `python/llm/example/CPU/Deepspeed-AutoTP/install.sh`
> 
> Update `pip install oneccl_bind_pt==2.1.100 -f https://developer.intel.com/ipex-whl-stable-xpu` to `pip install oneccl_bind_pt==2.1.100 --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/`

Need to add CN link as well, either in a separate tab or a note

**Comment by chtanch** [comment]
Applied changes in comments. 

Preview - https://chtanch-docs.readthedocs.io/en/update_pip_install/doc/LLM/Overview/install_gpu.html

File: python/llm/README.md (hunk 3 above)
Comment (hkvision) [reply python/llm/README.md:125]: This file hasn't been removed @chtanch

**Review by hkvision — APPROVED** [review]
LGTM. Will merge it after removing the README.

**Comment by hkvision** [comment]
> Applied changes in comments.
> 
> Preview - https://chtanch-docs.readthedocs.io/en/update_pip_install/doc/LLM/Overview/install_gpu.html

The documents look nice and no issue observed.

**Pull request merged**